capstone/src/test/java/com/google/sps/UpdateAssociationTest.java

```diff
@@ -0,0 +1,82 @@
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.language.v1.AnalyzeEntitySentimentRequest;
+import com.google.cloud.language.v1.Entity;
+import com.google.cloud.language.v1.EntityMention;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import java.util.ArrayList;
+import java.util.Arrays;
+import org.junit.Before;
+
+public class UpdateAssociationTest {
+
+  private LanguageServiceClient mockNLP;
+
+  @Before
+  public void setup() {
+    setupMockNLP();
+  }
+
+  private void setupMockNLP() {
+    mockNLP = mock(LanguageServiceClient.class);
+
+    when(mockNLP.analyzeEntitySentiment(any(AnalyzeEntitySentimentRequest.class)).getEntitiesList())
+        .thenReturn(new ArrayList<Entity>())
+        .thenReturn(Arrays.asList(getEntityOne(), getEntityThree()))
+        .thenReturn(
+            Arrays.asList(getEntityOne(), getEntityTwo(), getEntityThree())); // any subsequent call
+  }
+
+  private Entity getEntityOne() {
```

Comment:
Can you add a comment on when one would choose entity 1 vs 2 vs 3 (similar for mentions)?

Reply:
In this case there wasn't really a reason behind the different entities and mentions besides variety and demonstrating that you could mock cloud NLP and return different things.